src/TraceEvent/BPerf/BPerfEventSource.cs

```diff
@@ -113,6 +147,18 @@ public BPerfEventSource(string btlFilePath, TraceEventDispatcherOptions options,
             this.skipQPC = 0;
             this.endFileOffset = long.MaxValue;
             this.canResolveSymbols = !skipReadingUnreachableEvents;
+            this.DecompressDelegate = decompressDelegate;
+            if(this.DecompressDelegate == null)
+            {
+                foreach(var key in BPerfEventSource.BPerfDecompressDelegateDictionary.Keys)
```

Comment:
Each instance of `BPerfEventSource` is responsible for opening exactly one file.  Thus, it feels to me like the registration mechanism and dictionary are probably more than we need.  What do you think about allowing callers of the `BPerfEventSource` to specify a decompression delegate, and if it's not present, just default to the existing implementation?

Reply:
fixed, thanks for your time.